tests/matomo/test_matomo_pipeline.py

```diff
@@ -0,0 +1,175 @@
+        "date": "2020-01-01",
+        "period": "day",
+        "extra_params": {}
+    },
+    {
+        "resource_name": "sample_analytics_data2",
+        "methods": ["VisitsSummary.get", "API.getReportMetadata", "CustomReports.getCustomReport"],
+        "date": "2022-01-01",
+        "period": "day",
+        "extra_params": {"idCustomReport": 1}
+    }
+]
+QUERIES_START_DATE1 = [
+        {
+            "resource_name": "sample_analytics_data1",
+            "methods": ["VisitsSummary.get"],
+            "date": "2020-01-01",
+            "period": "day",
+            "extra_params": {}
+        }
+]
+QUERIES_START_DATE2 = [
+        {
+            "resource_name": "sample_analytics_data1",
+            "methods": ["VisitsSummary.get"],
+            "date": "2022-01-01",
+            "period": "day",
+            "extra_params": {}
+        }
+]
+ALL_TABLES_START_DATE = ["sample_analytics_data1_visits_summary_get"]
+REPORTS_SITE_ID = 3
+LIVE_EVENTS_SITE_ID = 3
+# dict containing the name of the tables expected in the db as keys and the number of rows expected as values
+ALL_TABLES_REPORTS = ["sample_analytics_data1_visits_summary_get", "sample_analytics_data1_api_get_report_metadata", "sample_analytics_data2_visits_summary_get",
+                      "sample_analytics_data2_api_get_report_metadata", "sample_analytics_data2_custom_reports_get_custom_report"]
+ALL_TABLES_EVENTS = ["visitors", "visits"]
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_reports(destination_name: str) -> None:
+    """
+    Checks to see that loading regular reports yields expected results.
+    :param destination_name: Name of the db the data is loaded to.
+    :returns: None
+    """
+
+    pipeline = _create_pipeline(destination_name=destination_name, dataset_name="matomo_dataset", include_reports=True, include_events=False, queries=QUERIES)
+    _check_pipeline_has_tables(pipeline, ALL_TABLES_REPORTS)
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_events(destination_name: str) -> None:
+    """
+    Checks to see that loading live events will yield expected results.
+    :param destination_name: Name of the db the data is loaded to.
+    :returns: None
+    """
+
+    pipeline = dlt.pipeline(destination=destination_name, full_refresh=True, dataset_name="matomo_dataset")
+    data_events = matomo_events(live_events_site_id=LIVE_EVENTS_SITE_ID, get_live_event_visitors=True)
+    info = pipeline.run(data_events)
+    assert_load_info(info)
+    _check_pipeline_has_tables(pipeline, ALL_TABLES_EVENTS)
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_incrementing_reports(destination_name: str) -> None:
+    """
+    Checks that incremental loading works for reports
+    @:param: destination_name - redshift/bigquery/postgres
+    """
+    # Define all end_dates to be used for incremental loading
+    first_end_date = pendulum.datetime(year=2020, month=1, day=1)
```

Comment:
where those dates are coming from? they should split the existing dataset in several parts

Reply:
Those are some random dates for the test data. With the demo matomo site every split has some data.